.github/workflows/ci.yml

```diff
@@ -0,0 +1,118 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  project-checks:
+    name: Project Checks
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: "stable"
+
+    - name: Install ltag
+      run: go install github.com/containerd/ltag@latest
+
+    - name: Run file header checks
+      run: ltag --check -t "./script/validate/template" -v
```

Comment:
Reused the mechanism I've used in containerd and other projects for file header checks unless there is a more preferred option?